python_modules/dagster/dagster/core/definitions/pipeline_sensor.py

```diff
@@ -335,7 +336,22 @@ def _wrapped_fn(context: SensorEvaluationContext):
                 run_records = context.instance.get_run_records(
                     filters=PipelineRunsFilter(run_ids=[event_log_entry.run_id])
                 )
-                check.invariant(len(run_records) == 1)
+
+                # skip if we couldn't find the right run
+                if len(run_records) != 1:
+                    # bc we couldn't find the run, we use the event timestamp as the approximate
+                    # run update timestamp
+                    approximate_update_timestamp = utc_datetime_from_timestamp(
+                        event_log_entry.timestamp
+                    )
+                    context.update_cursor(
+                        RunStatusSensorCursor(
+                            record_id=storage_id,
+                            update_timestamp=approximate_update_timestamp.isoformat(),
+                        ).to_json()
+                    )
+                    continue
```

Comment:
[1] here i chose to simply skip this iteration.
alternatives:
1. log the detailed skip reason in a `SkipReason`. but we will need to enable sensor evaluation yield both SkipReason and RunReaction at the same time, bc we would have both in one tick.
2. fire a `PipelineRunReaction` which passes the skip reason via the the check error. but similarly we will need to deal with `Job tick status is SUCCESS, but error was provided` as we would have both success RunReaction and error RunReaction in one tick.